soc/realtek/ameba/common/os_wrapper/include/os_wrapper_mutex.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2024 Realtek Semiconductor Corp.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __OS_WRAPPER_MUTEX_H__
+#define __OS_WRAPPER_MUTEX_H__
+
+#include <zephyr/kernel.h>
+
+#define MUTEX_WAIT_TIMEOUT 0xFFFFFFFFU
+
+/**
+ * @brief  mutex handle type
+ */
+typedef struct k_mutex *rtos_mutex_t;
+
+/**
+ * @brief  For Zephyr, map to k_mutex_init.
+ * @note   Usage example:
+ * Create:
+ *          rtos_mutex_t mutex_handle;
+ *          rtos_mutex_create(&mutex_handle);
+ * Give:
+ *          rtos_mutex_give(mutex_handle);
+ * Take:
+ *          rtos_mutex_take(mutex_handle, 100);
+ * Delete:
+ *          rtos_mutex_delete(mutex_handle);
+ * @param  pp_handle: The handle itself is a pointer, and the pp_handle means a pointer to the
+ * handle.
+ * @retval
+ */
+int rtos_mutex_create(rtos_mutex_t *pp_handle);
```

Comment:
@gmarull can you make heads or tails of this?

Reply:
Our driver requires semaphore services provided by the OS. By using this interface, our driver can support different operating systems.

Reply:
Why are the drivers not in zephyr in the `drivers` folder like every other vendor?

Reply:
Apologies for the misunderstanding. The code in the HAL repositories requires these APIs, and the review comment in the [HAL module request](https://github.com/zephyrproject-rtos/zephyr/issues/78084) stating ‘Zephyr glue code is NOT allowed in HAL repositories’ necessitates moving these APIs to the Zephyr repository.